src/platform/platform_worker.c

```diff
@@ -471,22 +563,39 @@ CxPlatRunExecutionContexts(
     } while (*EC != NULL);
 
     if (NextTime == 0) {
-        State->WaitTime = 0;
+        Worker->State.WaitTime = 0;
     } else if (NextTime != UINT64_MAX) {
-        uint64_t Diff = NextTime - State->TimeNow;
+        uint64_t Diff = NextTime - Worker->State.TimeNow;
         Diff = US_TO_MS(Diff);
         if (Diff == 0) {
-            State->WaitTime = 1;
+            Worker->State.WaitTime = 1;
         } else if (Diff < UINT32_MAX) {
-            State->WaitTime = (uint32_t)Diff;
+            Worker->State.WaitTime = (uint32_t)Diff;
         } else {
-            State->WaitTime = UINT32_MAX-1;
+            Worker->State.WaitTime = UINT32_MAX-1;
         }
     } else {
-        State->WaitTime = UINT32_MAX;
+        Worker->State.WaitTime = UINT32_MAX;
     }
 }
 
+uint32_t
+CxPlatWorkerPoolWorkerPoll(
+    _In_ QUIC_EXECUTION* Execution
+    )
+{
+    CXPLAT_WORKER* Worker = (CXPLAT_WORKER*)Execution;
+    Worker->State.ThreadID = CxPlatCurThreadID();
```

Comment:
TODO: Set this in all callbacks?

Reply:
Shouldn't be necessary, as we're not using this in api.c anymore, because we're a custom execution context.